fourm/data/modality_transforms.py

```diff
@@ -916,8 +1117,174 @@ def postprocess(self, sample):
         return sample
 
 
-class CaptionEmbTransform(AbstractTransform):
+class VideoTranscriptTransform(AbstractTransform):
+    def __init__(self, aligned_captions=True, no_aug=False):
+        # TODO: what are these args? Do we still need them?
+        self.aligned_captions = aligned_captions
+        self.no_aug = no_aug
+
+    def load(self, path) -> List[dict]:
+        # TODO: decide on the representation of the input description (how do we know which frames each caption maps to?)
+        # Caption can either be stored as .txt or .json.gz (in which case it's a list of dicts)
+        """
+        For now, assume we have something like a jsonlines of transcripts for each clip/sequence of frames in a video.:
+        [
+            {
+                "transcript": "here's a transcript",
+                "start_frame_index": 0,
+                "end_frame_index": 5,
+            },
+            {
+                "transcript": "here's another transcript",
+                "start_frame_index": 10,
+                "end_frame_index": 13,
+            } # Note that the transcript need not be consecutive in all frames, e.g., you can see a skip from frames 5-10 with no transcripts.
+        ]
+        """
+        with jsonlines.open(path, "r") as jsonl_f:
+            transcripts_per_clip = [obj for obj in jsonl_f]
+
+        return transcripts_per_clip
+
+    def preprocess(self, sample):
+        return sample
+
+    def image_augment(
```

Comment:
Precondition: assume val does not have any augmentations (thus list[dict])
postcondition: output is type list[dict]